src/writer/cluster.cpp

```diff
@@ -47,6 +47,7 @@ namespace writer {
 
 Cluster::Cluster(Compression compression)
   : compression(compression),
+    compressionLevel(0),
```

Comment:
This set the compressionLevel of the cluster but it seems that you never change it (`setCompressionLevel` is never called).
To avoid this kind of bug, it would be better to change the Cluster constructor's signature to add the compressionLevel.